lib/tiled/map.rb

```diff
@@ -89,6 +89,28 @@ def pixelheight
       return height * tileheight
     end
 
+    #gets the first object layer with the passed name of `layer_name`
+    #then looks in that layer for the first object that has the name `object_name`
+    #then returns that object
+    def object_by_name(layer_name, object_name)
+      
+      named_layer = @object_groups.reject do | layer |
+        layer.name != layer_name
+      end.first
+
+      raise ObjectLayerNotFound, "The object layer \"#{layer_name}\" was not found in map \"#{@path}\"" if named_layer.nil?
+      
+      the_object = named_layer.objects.reject do | object |
```

Comment:
`named_layer` will be a `Tiled::ObjectLayer` and has `find_by_name` method, so you can use it to find object by name:
```ruby
the_object = named_layer.find_by_name(object_name)
```
It will always return first object or nil if object is not found.